ee/secureenclavesigner/secureenclavesigner_darwin.go

```diff
@@ -0,0 +1,240 @@
+//go:build darwin
+// +build darwin
+
+package secureenclavesigner
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"os/user"
+	"strings"
+	"sync"
+
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/kolide/launcher/ee/allowedcmd"
+	"github.com/kolide/launcher/ee/consoleuser"
+)
+
+const (
+	CreateKeyCmd = "create-key"
+)
+
+type opt func(*SecureEnclaveSigner)
+
+type SecureEnclaveSigner struct {
+	uidPubKeyMap         map[string]*ecdsa.PublicKey
+	pathToLauncherBinary string
+	persister            persister
+	slogger              *slog.Logger
+	mux                  *sync.Mutex
+}
+
+// persister is an interface that allows the secure enclave signer to persist
+// keys it creates, this is needed since we may need to create a new key anytime
+// a new user logs in on macos
+type persister interface {
+	Persist([]byte) error
+}
+
+func New(slogger *slog.Logger, persister persister, opts ...opt) (*SecureEnclaveSigner, error) {
+	ses := &SecureEnclaveSigner{
+		uidPubKeyMap: make(map[string]*ecdsa.PublicKey),
+		persister:    persister,
```

Comment:
instead of expecting the calling code to unmarshall a json in to this struct, mabye we should just pull the data in new?